frontend/app/src/app/_services/chatroom.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { catchError, map, tap } from 'rxjs/operators';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Chatroom } from '../_models/chatroom'
+import { Message } from '../_models/message';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ChatroomService {
+
+  private chatroomsUrl = 'http://localhost:8000/messages';
```

Comment:
Don't use hardcoded address. You need to put `apiEndpoint: "127.0.0.1:8000"` into some config